lib/router/transition.js

```diff
@@ -58,7 +70,7 @@ function Transition(router, intent, state, error) {
       this.pivotHandler = handlerInfo.handler;
     }
 
-    this.sequence = Transition.currentSequence++;
+    this.sequence = router.currentSequence++;
     this.promise = state.resolve(checkForAbort, this)['catch'](function(result) {
```

Comment:
Hmm. We definitely need to figure out the flags to know which method to use, do you have suggestions that would allow us to accomplish both objectives?